src/components/triggersAndActions.ts

```diff
@@ -0,0 +1,176 @@
+import path from 'path';
+import getAppRootDir from './getAppRoot';
+
+const schedule = require('node-schedule');
+
+export interface DatabaseRecord {
+  [key: string]: any;
+  code: string;
+}
+
+export interface DatabaseResult {
+  rows: DatabaseRecord[];
+}
+
+export interface TriggerPayload {
+  id: number;
+  trigger: string;
+  table: string;
+  record_id: number;
+}
+
+export interface ActionPayload {
+  id: number;
+  code: string;
+  parameters: { [key: string]: any };
+}
+
+export interface Action {
+  code: string;
+  path: string;
+  name: string;
+  trigger: string;
+  condition: { [key: string]: any };
+  parameter_queries: { [key: string]: any };
+}
+
+const pluginFolder = path.join(getAppRootDir(), 'plugins'); //Change this to config file
+
+// Load actions from Database at server startup
+export const loadActions = async function (
+  client: { [key: string]: Function },
+  actionLibrary: { [key: string]: Function }
```

Comment:
Okay, I've defined ActionLibrary as an Interface. Left the client one for now, because it should be defined by the package it came with `node-postgres`, but doesn't seem to be defined. There's probably a community-defined type for it somewhere, but not sure how to import and work with this yet. Leaving it as is as a reminder to re-look at it (I'll make an issue too).